drivers/flash/flash_stm32wbx.c

```diff
@@ -62,15 +67,105 @@ static int write_dword(struct device *dev, off_t offset, uint64_t val)
 		return -EIO;
 	}
 
-	/* Set the PG bit */
-	regs->CR |= FLASH_CR_PG;
+	/* Implementation of STM32 AN5289, proposed in STM32WB Cube Application
+	 * BLE_RfWithFlash
+	 * https://github.com/STMicroelectronics/STM32CubeWB/tree/master/Projects/P-NUCLEO-WB55.Nucleo/Applications/BLE/BLE_RfWithFlash
+	 */
+
+	do {
+		/**
+		 * When the PESD bit mechanism is used by CPU2 to protect its
+		 * timing, the PESD bit should be polled here.
+		 * If the PESD is set, the CPU1 will be stalled when reading
+		 * literals from an ISR that may occur after the flash
+		 * processing has been requested but suspended due to the PESD
+		 * bit.
+		 *
+		 * Note: This code is required only when the PESD mechanism is
+		 * used to protect the CPU2 timing.
+		 * However, keeping that code make it compatible with both
+		 * mechanisms.
+		 */
```

Comment:
This looks like copy from https://github.com/STMicroelectronics/STM32CubeWB/blob/master/Projects/P-NUCLEO-WB55.Nucleo/Applications/BLE/BLE_RfWithFlash/Core/Src/flash_driver.c#L185, code which is not Apache licensed. Not sure if that could be a problem? @erwango

Reply:
This is indeed an issue. We're triggering discussions to sort this out.

Reply:
After internal FOSS Out process, agreement is reached to allow this code to be published with Apache licence.
I simply added STMicroelectronics copyright.